app/controllers/recipes_controller.rb

```diff
@@ -0,0 +1,16 @@
+class RecipesController < ApplicationController
+
+  def show
+    @recipe = Recipe.find(params[:id])
+    @ingredients = @recipe.ingredients
```

Comment:
While this isn’t wrong, it’s not necessary or conventional to create these additional instance variables as they are attributes for your first instance var. You can call `@recipe.ingredients` or `@recipe.total_cost` in the view and that is not a violation of MVC. What would violate MVC would be doing an AR query like `@recipe.find` or `@recipes.count` in the view